addons/addon-raas-s3-copy/packages/s3-synchronizer/src/s3-download.go

```diff
@@ -124,34 +282,49 @@ func getObjectsAll(
 			continue
 		}
 
-		if debug == true {
-			log.Printf("%v -> %v\n", *item.Key, destFilePath)
-		}
-
 		// Ensure the directory exists
 		os.MkdirAll(filepath.Dir(destFilePath), 0775)
 
-		downloader := s3manager.NewDownloader(sess, func(d *s3manager.Downloader) {
-			d.PartSize = 100 * 1024 * 1024
-			d.Concurrency = concurrency
-		})
+		shouldDownload := true
+
+		// Note that we cannot use os.IsExist(fileError) to check for file's existence
+		// os.IsExist and os.IsNotExist are error checker functions and only work when error is not nil
+		// The correct way to check if file exists is using !os.IsNotExist(fileError)
+		if _, fileError := os.Stat(destFilePath); !os.IsNotExist(fileError) {
+			// If the file has not changed in S3 since last download then skip downloading it
+			shouldDownload = hasFileChangedInS3(item)
```

Comment:
If the study was read only, should we disallow any modifications/writes to this local directory just like Goofys does. I know we aren't implementing Goofys here but thought we should discuss this out. Another option would be to have another flag which doesn't force the download if the file was modified and leaves it as it is. Do you know what is the expected behavior or what we have is what is desirable?

Reply:
In case of ReadOnly, the S3 changes should win. The function `"hasFileChangedInS3"` checks if the file has changed in S3 and needs re-download. This is not about changes in local directory.

Reply:
Makes sense. Just for my clarification though- is the local study mount/directory read-only though? Or existing files can be modified or new files be added by the local user.

Reply:
No the directory will not be ReadOnly on the file system. User's changes will be overwritten by the synchronizer in case of read-only. Cannot make the directory/files read-only on the file system because the researchers are also logging in as the "Administrator" user - i.e., the same user the synchronizer is running as. Making them read-only will make the synchronizer also fail. 
Will add a separate ticket to investigate this further.

Reply:
Separate ticket sounds good. Once you do that, mark this conversation as resolved.

Reply:
Done. Created ticket.